shell/common/shell_unittests.cc

```diff
@@ -2014,5 +2014,61 @@ TEST_F(ShellTest, OnServiceProtocolEstimateRasterCacheMemoryWorks) {
   DestroyShell(std::move(shell));
 }
 
+TEST_F(ShellTest, DiscardLayerTreeOnResize) {
+  auto settings = CreateSettingsForFixture();
+
+  std::shared_ptr<ShellTestExternalViewEmbedder> external_view_embedder;
+
+  SkISize wrong_size = SkISize::Make(400, 100);
+  SkISize expected_size = SkISize::Make(400, 200);
+
+  fml::AutoResetWaitableEvent end_frame_latch;
+
+  auto end_frame_callback = [&](bool, fml::RefPtr<fml::RasterThreadMerger>) {
+    end_frame_latch.Signal();
+  };
+
+  external_view_embedder = std::make_shared<ShellTestExternalViewEmbedder>(
```

Comment:
nit: declare this variable here rather than line:2020. See https://google.github.io/styleguide/cppguide.html#Local_Variables

Reply:
Yeah. It was accessed from the callback before. I'll move it.